src/inference/gibbs.jl

```diff
@@ -36,25 +36,38 @@ end
 alg = Gibbs(HMC(0.2, 3, :v1), PG(20, :v2))
 ```
 
+One can also pass the number of iterations for each Gibbs component using the following syntax:
+- `alg = Gibbs((HMC(0.2, 3, :v1), n_hmc), (PG(20, :v2), n_pg))`
+- `alg = Gibbs(HMC(0.2, 3, :v1) => n_hmc, PG(20, :v2) => n_pg)`
+where `n_hmc` and `n_pg` are the number of iterations for HMC and PG in each Gibbs iteration.
+
 Tips:
 - `HMC` and `NUTS` are fast samplers, and can throw off particle-based
 methods like Particle Gibbs. You can increase the effectiveness of particle sampling by including
 more particles in the particle sampler.
 """
-struct Gibbs{space, A<:Tuple} <: InferenceAlgorithm
+struct Gibbs{space, A<:Tuple, B<:Tuple} <: InferenceAlgorithm
     algs::A   # component sampling algorithms
-
-    function Gibbs{space, A}(algs::A) where {space, A<:Tuple}
+    iterations::B
+    function Gibbs{space, A, B}(algs::A, iterations::B) where {space, A<:Tuple, B<:Tuple}
         all(isgibbscomponent, algs) || error("all algorithms have to support Gibbs sampling")
-        return new{space, A}(algs)
+        return new{space, A, B}(algs, iterations)
     end
 end
 
-function Gibbs(algs...)
+function Gibbs(algs::InferenceAlgorithm...)
```

Comment:
This also defines `Gibbs`. To ensure that at least one algorithm is specified, we should use

```suggestion
function Gibbs(alg::InferenceAlgorithm, algs::Vararg{InferenceAlgorithm,N}) where {N}
```

Reply:
The exisiting `Gibbs` constructor has to be parametric (i.e. `Gibbs{space, A, B}(algs, iterations)`) so I don't think it defines `Gibbs`.

Reply:
A simple example that shows it does:
```julia
julia> struct Gibbs{space, A<:Tuple, B<:Tuple}
           algs::A
           iterations::B
           function Gibbs{space,A,B}(algs::A, iterations::B) where {space,A<:Tuple,B<:Tuple}
               return new{space,A,B}(algs, iterations)
           end
       end

julia> function Gibbs(algs::String...)
           iterations = ntuple(_ -> 1, Val(length(algs)))
           return Gibbs{:x,typeof(algs),typeof(iterations)}(algs, iterations)
       end
Gibbs

julia> Gibbs()
Gibbs{:x, Tuple{}, Tuple{}}((), ())
```

It's a bit annoying but every time you use `Vararg`s in Julia, you also define a method without any `Vararg`s and potentially redefine existing functions.

Reply:
And the definition below redefines it:
```julia
julia> function Gibbs(args::Pair{Real,Int}...)
           algs = map(first, args)
           iterations = map(last, args)
           return Gibbs{:x,typeof(algs),typeof(iterations)}(algs, iterations)
       end
Gibbs

julia> Gibbs()
ERROR: MethodError: Gibbs() is ambiguous. Candidates:
  Gibbs(algs::String...) in Main at REPL[3]:1
  Gibbs(args::Pair{Real, Int64}...) in Main at REPL[5]:1
Possible fix, define
  Gibbs()
Stacktrace:
 [1] top-level scope
   @ REPL[6]:1
```